### PR TITLE
[RA-12] Memory cache + response caching, inspired in insight's scheme

### PR DESCRIPTION
**Pull request by dario-ramos**

### CODE REVIEW

File: bitprim.insight/Controllers/BlockController.cs [hunk 1]
```diff
@@ -51,6 +55,12 @@ public ActionResult Simulate()
             }
 
             Utils.CheckIfChainIsFresh(chain_, config_.AcceptStaleRequests);
+
+            JsonResult cachedBlockJson;
+            if(memoryCache_.TryGetValue("block" + hash, out cachedBlockJson))
```
Comment (mdallago) [inline bitprim.insight/Controllers/BlockController.cs]: Why use memoryCache_ and ResponseCache ? How much time the objects are held in memoryCache ?

Comment (dario-ramos) [reply bitprim.insight/Controllers/BlockController.cs]: Because insight uses those two mechanisms (via lru-cache for memory, and Angular for responses)

Comment (dario-ramos) [reply bitprim.insight/Controllers/BlockController.cs]: Memory cache entries will only be evicted due to memory pressure, with LRU criteria, AFAIK

Comment (mdallago) [reply bitprim.insight/Controllers/BlockController.cs]: we need to specify a limit in the memory use by the cache.

**Pull request merged**